lib/sdk/core/observer.js

```diff
@@ -41,13 +41,12 @@ const ObserverDelegee = Class({
     this.delegate = delegate;
   },
   QueryInterface: function(iid) {
-    const isObserver = iid.equals(Ci.nsIObserver);
-    const isWeakReference = iid.equals(Ci.nsISupportsWeakReference);
+    if (iid.equals(Ci.nsIObserver) ||
+        iid.equals(Ci.nsISupportsWeakReference) ||
+        iid.equals(Ci.nsISupports))
+      return this;
```

Comment:
We can do but I suspect it's not a problem, this is basically the standard form for QueryInterface: http://mxr.mozilla.org/mozilla-central/source/js/xpconnect/loader/XPCOMUtils.jsm#339